base_decorator.rb

```diff
@@ -0,0 +1,14 @@
+require_relative 'nameable'
+
+class BaseDecorator < Nameable
+  attr_accessor :nameable
+
+  def initilize(nameable)
```

Comment:
- [ ] You have a typo error here. This should be `initialize`. You are missing an "a".